tests/t_system.nim

```diff
@@ -12,6 +12,36 @@ proc execSystemTests*(runnable: bool)=
         let bgBitmap = playdate.graphics.newBitmap(400, 240, kColorBlack)
         playdate.system.setMenuImage(bgBitmap, 0)
 
+    test "runCatching typed int with exception":
+      if(runnable):
+        var result: int
+        result = runCatching(proc(): int = raise newException(ValueError, "test Exveption with int return type"))
+        assert(result == 0)
+
+    test "runCatching untyped with exception":
+      if runnable:
```

Comment:
Do we expect this to be run on the headless test? the whole sparsemap suite and System APi suite seems to be missing in the logs https://github.com/samdze/playdate-nim/actions/runs/14674842933/job/41189152102?pr=109